src/scripts/setup.sh

```diff
@@ -12,6 +12,25 @@ Install() {
     if [ "$UID" = "0" ]; then export SUDO=''; else export SUDO='sudo'; fi
     $SUDO find /etc/datadog-agent/conf.d/ -iname "*.yaml.default" -delete
     $SUDO service datadog-agent start
+
+    set +e
+    attempts=0
+    $SUDO datadog-agent health
+    exit_code=$?
+    until [[ $attempts -ge 10 ||  $exit_code -eq 0 ]]; do
```

Comment:
I think you can put `$SUDO datadog-agent health` inside the until condition and it will always be the last command called so you do not need to save the exit code or have the first call out of the loop.

It'd need testing though but it's the usual usage for `until` that I've seen

Reply:
This is to avoid waiting if it works the first time around mainly.

Reply:
It doesn't. And your current code works the same way anyway :)

Reply:
> It doesn't. And your current code works the same way anyway :)

Sorry I don't think I follow. What do you mean with it doesn't having that block outside the loop means that if the first command returns with exit code 0 it skips the loop and saves and extra 10 seconds. 

If I remove this it will always do the first sleep won't it?

Reply:
If you put the command in the `until` line, it would still skip the loop

Reply:
Ok, sorry I misunderstood before, I thought you meant inside the until block and not literally in the condition evaluation. 

I'll give that a try.

Reply:
I ended up spending more time that I care to admit to get it working but it does now. I really have to brush up on my bash script.